static/app/utils/queryClient.tsx

```diff
@@ -17,6 +18,24 @@ type UseQueryOptions<TQueryFnData, TError, TData, TQueryKey extends QueryKey> =
   'queryKey' | 'queryFn'
 >;
 
+interface UseMutationOptionsWithApiOptions<TData, TError, TVariables, TContext>
+  extends Omit<
+    reactQuery.UseMutationOptions<TData, TError, TVariables, TContext>,
+    'mutationFn'
+  > {
+  api: {
+    method: Exclude<APIRequestMethod, 'GET'>;
+    url: string;
+    data?: any;
+    query?: Record<string, any>;
+  };
+  mutationFn?: never;
```

Comment:
I never used the type never :) 

can this type also be the following?

```suggestion
  mutationFn?: () => Promise<void>;
```

Reply:
The reason I used `never` here was to prevent someone from passing _both_ `api` and `mutationFn`. So this would error:

```ts
useMutation({ api: {...}, mutationFn: () => {...} })
```

But these won't:

```ts
useMutation({ api: {...} })
useMutation({ mutationFn: () => {...} })
```